keras_nlp/utils/preset_utils.py

```diff
@@ -155,6 +158,101 @@ def save_to_preset(
             metadata_file.write(json.dumps(metadata, indent=4))
 
 
+def _validate_tokenizer(
+    preset, config_file="tokenizer.json", allow_incomplete=False
+):
+    config_path = get_file(preset, config_file)
+    if not os.path.exists(config_path):
+        if allow_incomplete:
+            logging.warning(
+                f"Tokenizer is missing from the preset directory {preset}."
+            )
+            return
+        else:
+            raise FileNotFoundError(
+                f"Tokenizer is missing from the preset directory {preset}."
+                "If you wish to upload model without a tokenizer, set `allow_incomplete=True`."
+            )
+    with open(config_path) as config_file:
+        config = json.load(config_file)
+    layer = keras.saving.deserialize_keras_object(config)
+
+    tokenizer = get_tokenizer(layer)
+    if tokenizer and config["assets"]:
+        for asset in config["assets"]:
+            asset_path = os.path.join(preset, asset)
+            if not os.path.exists(asset_path):
+                raise FileNotFoundError(
+                    f"Asset {asset} doesn't exist in the preset direcotry {preset}."
+                )
+        config_dir = os.path.dirname(config_path)
+        asset_dir = os.path.join(config_dir, TOKENIZER_ASSET_DIR)
+        tokenizer.load_assets(asset_dir)
+    else:
+        raise ValueError("Tokenizer or its asset are missing or invalid.")
+
+
+def _validate_backbone(preset, config_file="config.json"):
+    config_path = get_file(preset, config_file)
+    with open(config_path) as config_file:
+        config = json.load(config_file)
+    # Check if backbone is deserializable.
+    keras.saving.deserialize_keras_object(config)
+
+    if config["weights"]:
+        weights_path = os.path.join(preset, config["weights"])
+        if not os.path.exists(weights_path):
+            raise FileNotFoundError(
+                f"The weights file doesn't exist in preset directory {preset} ."
+            )
+    else:
+        raise ValueError("there is no wieghts config!")
```

Comment:
Let's make this more actionable. Something like...
```
No weights listed in `config.json`. Make sure to use `save_to_preset()` which adds additional data to a serialized Keras object.
```

Also capitalization, wieghts -> weights.

Reply:
Thanks! Done!